### PR TITLE
An 2663/orca lp actions

### PR DESCRIPTION
**Pull request by tarikceric**
Liquidity Pool Actions performed on Orca Dex

Models overview:
- `Liquidity_pool_events_orca`: all events related to liquidity_pools
  - Events are identified as `deposit`/`withdraw`
  - Whirlpool deposits/withdraws have the same structure, so they are identified as `whirlpool_unknown` and are filtered further in downstream models
    - Whirlpool also has `whirlpool_fee_withdraw` actions, which are eventually counted as `withdraws` downstream

- Transfers: `Pool_transfers_orca_whirlpool` and `pool_transfers_orca_non_whirlpool`
  - Token transfers for deposits/withdraws with separate models for whirlpool and non-whirlpool program_ids

- Mints: `mints_orca_whirlpool` and `mints_orca_non_whirlpool`

- Burns: `burns_orca_whirlpool` and `burns_orca_non_whirlpool`

Testing:
- The TVL values for a liquidity pools can be compared to the TVL values found on Solscan
  - IE For a stable-coin pair, calculate the sum of withdraws + deposits, and this should equal the TVL within Solscan
`select 'deposit', sum(amount) from solana_dev.silver.pool_transfers_orca_whirlpool
where succeeded
and liquidity_pool_address = 'GLNvG5Ly4cK512oQeJqnwLftwfoPZ4skyDwZWzxorYQ9'
and mint in ('7kbnvuGBxxj8AG9qp8Scn56muWGaRaFqxg1FsRp3PaFT','EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v') and ACTION = 'deposit'
union
select 'withdraw', sum(amount) from solana_dev.silver.pool_transfers_orca_whirlpool
where succeeded
and liquidity_pool_address = 'GLNvG5Ly4cK512oQeJqnwLftwfoPZ4skyDwZWzxorYQ9'
and mint in ('7kbnvuGBxxj8AG9qp8Scn56muWGaRaFqxg1FsRp3PaFT','EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v') and ACTION = 'withdraw';` 

- For a pool token, the total supply can be calculated (mints - burns) and compared to the value on Solscan
`select 'mints',sum(amount) from solana_dev.silver.mints_orca_non_whirlpool
where succeeded
and mint = 'H2uzgruPvonVpCRhwwdukcpXK8TG17swFNzYFr2rtPxy'
union
select 'burns',sum(amount) from solana_dev.silver.burns_orca_non_whirlpool
where succeeded
and mint = 'H2uzgruPvonVpCRhwwdukcpXK8TG17swFNzYFr2rtPxy';`

### CODE REVIEW

File: models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                inner_instruction_program_ids
+            )
+            OR ARRAY_CONTAINS(
+                'DjVE6JNiYqPL2QXyCUUh8rNjHrbz9hXHNYt99MQ59qw1' :: variant,
+                inner_instruction_program_ids
+            )
+        )
+
+{% if is_incremental() %}
+AND _inserted_timestamp >= (
```
Comment (desmond-hui) [inline models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql:36]: got error when trying to do incremental 

` Completed with 1 error and 0 warnings:
20:57:07  
20:57:07  Database Error in model silver__liquidity_pool_events_orca (models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql)
20:57:07    100090 (42P18): Duplicate row detected during DML action
20:57:07    Row Values: [redacted]
20:57:07    compiled Code at target/run/solana_models/models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql`

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql (hunk 1 above)
Comment (tarikceric) [reply models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql:36]: issue with my unique_key not being distinct enough -- added 'index' and 'inner_index' to it

Comment (desmond-hui) [reply models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql:36]: unique key columns cannot have `null` values, it will not merge properly

Comment (tarikceric) [reply models/silver/liquidity_pool/silver__liquidity_pool_events_orca.sql:36]: inner_index has been set to -1 instead of null

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**